src/main/java/com/exasol/adapter/dialects/exasol/ExasolLocalSqlGenerationVisitor.java

```diff
@@ -0,0 +1,42 @@
+package com.exasol.adapter.dialects.exasol;
+
+import com.exasol.adapter.AdapterException;
+import com.exasol.adapter.dialects.SqlDialect;
+import com.exasol.adapter.dialects.rewriting.SqlGenerationContext;
+import com.exasol.adapter.metadata.DataType;
+import com.exasol.adapter.sql.SqlColumn;
+/**
+ * This class generates SQL queries for the {@link ExasolSqlDialect}. It is a 'specialisation' for LOCAL mode, based on {@link ExasolSqlGenerationVisitor}.
+ */
+public class ExasolLocalSqlGenerationVisitor extends ExasolSqlGenerationVisitor{
+    /**
+     * Creates a new instance of the {@link ExasolLocalSqlGenerationVisitor}.
+     *
+     * @param dialect {@link ExasolSqlDialect} dialect
+     * @param context SQL generation context
+     */
+    ExasolLocalSqlGenerationVisitor(SqlDialect dialect, SqlGenerationContext context) {
+        super(dialect, context);
+    }
+    @Override
+    public String visit(final SqlColumn column) throws AdapterException {
+        ExasolSqlDialect exasolSqlDialect = new ExasolSqlDialect(null,null);
+        String tablePrefix = "";
+        if (column.hasTableAlias()) {
+            tablePrefix = exasolSqlDialect.applyQuote(column.getTableAlias())
+                    + exasolSqlDialect.getTableCatalogAndSchemaSeparator();
+        } else if ((column.getTableName() != null) && !column.getTableName().isEmpty()) {
+            tablePrefix = exasolSqlDialect.applyQuote(column.getTableName())
+                    + exasolSqlDialect.getTableCatalogAndSchemaSeparator();
+        }
```

Comment:
Extract this into a method

Reply:
Can you clarify this a bit further? Contacting you on slack.